hdlparse/vhdl_parser.py

```diff
@@ -6,748 +6,829 @@
+    (r'/\*', 'block_comment', 'block_comment'),
+    (r'--.*\n', None),
+  ],
+  'entity': [
+    (r'generic\s*\(', None, 'generic_list'),
+    (r'port\s*\(', None, 'port_list'),
+    (r'end\s+\w+\s*;', 'end_entity', '#pop'),
+    (r'/\*', 'block_comment', 'block_comment'),
+    (r'--.*\n', None),
+  ],
+  'architecture': [
+    (r'end\s+\w+\s*;', 'end_arch', '#pop'),
+    (r'/\*', 'block_comment', 'block_comment'),
+    (r'type\s+(\w+)\s*is', 'type', 'type_decl'),
+    (r'--.*\n', None),
+  ],
+  'generic_list': [
+    (r'\s*(\w+)\s*', 'generic_param'),
+    (r'\s*,\s*', None),
+    (r'\s*:\s*', None, 'generic_param_type'),
+    (r'--#(.*)\n', 'metacomment'),
+    (r'/\*', 'block_comment', 'block_comment'),
+    (r'--.*\n', None),
+  ],
+  'generic_param_type': [
+    (r'\s*(\w+)[ \t\r\f\v]*', 'generic_param_type'),
+    (r'\s*;\s*', None, '#pop'),
+    (r"\s*:=\s*([\w']+)", 'generic_param_default'),
+    (r'\)\s*;\s*--(.*)\n', 'line_comment', '#pop:2'),
+    (r'\n\s*\)\s*;\s*--(.*)\n', 'generic_list_comment', '#pop:2'),
+    (r'\n\s*', None),
+    (r'\)\s*;', 'end_generic', '#pop:2'),
+    (r'--#(.*)\n', 'metacomment'),
+    (r'/\*', 'block_comment', 'block_comment'),
+    (r'--.*\n', None),
+  ],
+  'port_list': [
+    (r'\s*(\w+)\s*', 'port_param'),
+    (r'\s*,\s*', None),
+    (r'\s*:\s*', None, 'port_param_type'),
+    (r'--#\s*{{(.*)}}\n', 'section_meta'),
+    (r'--#(.*)\n', 'metacomment'),
+    (r'/\*', 'block_comment', 'block_comment'),
+    (r'--(.*)\n', 'line_comment'),
+  ],
+  'port_param_type': [
+    (r'\s*(in|out|inout|buffer)\s+(\w+)\s*\(', 'port_array_param_type', 'array_range'),
+    (r'\s*(in|out|inout|buffer)\s+(\w+)[ \t\r\f\v]*', 'port_param_type'),
+    (r'\s*;\s*', None, '#pop'),
+    (r"\s*:=\s*([\w']+)", 'port_param_default'),
+    (r'--(.*)\n', 'line_comment'),
+    (r'\)\s*;\s*--(.*)\n', 'line_comment', '#pop:2'),
+    (r'\n\s*\)\s*;\s*--(.*)\n', 'port_list_comment', '#pop:2'),
+    (r'\n\s*', None),
+    (r'\)\s*;', 'end_port', '#pop:2'),
+    (r'--#(.*)\n', 'metacomment'),
+    (r'/\*', 'block_comment', 'block_comment'),
+  ],
+  'array_range': [
+    (r'\(', 'open_paren', 'nested_parens'),
+    (r'\s*([\w\+\-\*/\s]+)(\s+(?:down)?to)\s+([\w\+\-\*/\s]+)', 'array_range_val'),
+    (r'\)', 'array_range_end', '#pop'),
+  ],
+  'nested_parens': [
+    (r'\(', 'open_paren', 'nested_parens'),
+    (r'\s*([\w\+\-\*/\s]+)(\s+(?:down)?to)\s+([\w\+\-\*/\s]+)', 'array_range_val'),
+    (r'\)', 'close_paren', '#pop'),
+  ],
+  'block_comment': [
+    (r'\*/', 'end_comment', '#pop'),
+  ],
 }
 
 VhdlLexer = MiniLexer(vhdl_tokens, flags=re.MULTILINE | re.IGNORECASE)
 
 
 class VhdlObject(object):
-    """Base class for parsed VHDL objects
-
-    Args:
-      name (str): Name of the object
-      desc (str): Description from object metacomments
-    """
-
-    def __init__(self, name, desc=None):
-        self.name = name
-        self.kind = 'unknown'
-        self.desc = desc
+  '''Base class for parsed VHDL objects
 
+  Args:
```

Comment:
This is not a proper way to document arguments in Python.  
Please use syntax that can be analyzed by Sphinx.

(Can be done in a new PR. If so, please derive an issue to track the work and the PR.)

Reply:
Forwarded to #5.